src/Sign.Core/CertificateServices/CertificateStoreService.cs

```diff
@@ -28,25 +30,33 @@ internal sealed class CertificateStoreService : ISignatureAlgorithmProvider, ICe
 
         internal CertificateStoreService(
             IServiceProvider serviceProvider,
-            string sha1Thumbprint,
+            string certificateFingerprint,
+            HashAlgorithmName certificateFingerprintAlgorithm,
             string? cryptoServiceProvider,
             string? privateKeyContainer,
             string? certificatePath,
             string? certificatePassword,
             bool isPrivateMachineKeyContainer)
         {
-            if (string.IsNullOrEmpty(sha1Thumbprint))
-            {
-                throw new ArgumentException(Resources.ValueCannotBeEmptyString, nameof(sha1Thumbprint));
-            }
+            ArgumentException.ThrowIfNullOrEmpty(certificateFingerprint, nameof(certificateFingerprint));
 
-            _sha1Thumbprint = sha1Thumbprint;
+            _certificateFingerprint = certificateFingerprint;
+            _certificateFingerprintAlgorithm = certificateFingerprintAlgorithm;
             _cryptoServiceProvider = cryptoServiceProvider;
             _privateKeyContainer = privateKeyContainer;
             _isPrivateMachineKeyContainer = isPrivateMachineKeyContainer;
             _certificatePath = certificatePath;
             _certificatePassword = certificatePassword;
 
+            _hasher = _certificateFingerprintAlgorithm.Name switch
+            {
+                "SHA256" => SHA256.Create(),
+                "SHA384" => SHA384.Create(),
+                "SHA512" => SHA512.Create(),
+
+                _ => throw new NotSupportedException(Resources.VSIXSignToolUnsupportedAlgorithm)
```

Comment:
This will go away with my other suggested change in this file.  However, I wanted to comment on this.

I see the `CertificateStoreService` as being general purpose.  Yes, VSIX signing uses it, but anything should be able to use it.  The name of the resource string implies a VSIX restriction is being respected here.  (The message itself is not VSIX-specific.)

Ideally, at a global level, we should be able to allow or disallow specific algorithms, but enable the broadest set of options by default, while individual signature formatters (e.g.:  NuGet, VSIX, etc.) should be able to indicate they do or do not support specific algorithms.

This location is completely independent of signature format, but the resource string indicates otherwise.